src/main/java/com/google/starfish/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,195 @@
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+  // All database credentials are stored in environment variables
+  private static final String CLOUD_SQL_CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASS = System.getenv("DB_PASS");
+  private static final String DB_NAME = System.getenv("DB_NAME");
+
+  private final int MILLISECONDS_IN_SECOND = 1000;
+  private final int SECONDS_IN_MINUTE = SECOND * 60;
+
+  /** Configures database connection, opens pool, and returns pool as DataSource. */
+  private DataSource createConnectionPool() {
+    // The configuration object specifies behaviors for the connection pool.
+    HikariConfig config = new HikariConfig();
+
+    config.setJdbcUrl(String.format("jdbc:mysql:///%s", DB_NAME));
+    config.setUsername(DB_USER);
+    config.setPassword(DB_PASS);
+
+    config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.mysql.SocketFactory");
+    config.addDataSourceProperty("cloudSqlInstance", CLOUD_SQL_CONNECTION_NAME);
+
+    // maximumPoolSize limits the total number of concurrent connections this pool will keep.
+    config.setMaximumPoolSize(5);
+
+    // minimumIdle is the minimum number of idle connections Hikari maintains in the pool.
+    config.setMinimumIdle(5);
+
+    // setConnectionTimeout is the maximum number of milliseconds to wait for a connection checkout.
+    config.setConnectionTimeout(MILLISECONDS_IN_SECOND * 10);
+
+    // idleTimeout is the maximum amount of time a connection can sit in the pool. Connections that
+    // sit idle for this many milliseconds are retried if minimumIdle is exceeded.
+    config.setIdleTimeout(SECONDS_IN_MINUTE * 10);
+
+    // maxLifetime is the maximum possible lifetime of a connection in the pool. Connections that
+    // live longer than this many milliseconds will be closed and reestablished between uses.
+    config.setMaxLifetime(SECONDS_IN_MINUTE * 30);
+
+    // Initialize the connection pool using the configuration object.
+    DataSource pool = new HikariDataSource(config);
+
+    return pool;
+  }
+
+  private void createLabelsTable(DataSource pool) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "CREATE TABLE IF NOT EXISTS labels ( "
+              + "title VARCHAR(255) PRIMARY KEY,"
+              + "type ENUM('School', 'Course', 'Other') NOT NULL );";
```

Comment:
I'm sure you've thought of this, but I'm hoping there's a way to use sql_tables.sql you merged earlier to simplify the java code. Right now it doesn't seem like this file depends on the .sql at all.

Reply:
Yep, we should be able to simplify the java code! Was thinking this would be a later thing but we should be able to do what this SO post suggests: https://stackoverflow.com/questions/1044194/running-a-sql-script-using-mysql-with-jdbc